.github/workflows/nx-dependency-update.yaml

```diff
@@ -0,0 +1,76 @@
+name: nx Dependency Update
+
+on:
+  pull_request:
+    branches:
+      - main
+    types:
+      - opened
+      - reopened
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  update-nx-dependencies:
+    name: Update nx dependencies and apply migrations
+    runs-on: ubuntu-latest
+    if: contains(github.event.pull_request.title, 'bump nx from')
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+
+      - name: Setup git user
+        uses: fregante/setup-git-user@v2
+
+      # Ignore nx migrations.json, otherwise it would be committed with the first executed migration
+      - name: Ignore nx migrations.json
+        run: echo "migrations.json" >> .git/info/exclude
+
+      - name: Extract source nx version
+        id: source-nx-version
+        # yamllint disable-line rule:line-length
+        run: echo "version=$(echo "${{ github.event.pull_request.title }}" | sed -n 's/.*from \([^ ]*\).*/\1/p')" >> "$GITHUB_OUTPUT"
+
+      - name: Extract target nx version
+        id: target-nx-version
+        # yamllint disable-line rule:line-length
+        run: echo "version=$(echo "${{ github.event.pull_request.title }}" | sed -n 's/.*to \([^ ]*\).*/\1/p')" >> "$GITHUB_OUTPUT"
```

Comment:
There is a https://github.com/dependabot/fetch-metadata action which can pull all the info instead of parsing it from the PR title.

See also https://docs.github.com/en/code-security/dependabot/working-with-dependabot/automating-dependabot-with-github-actions for some docs about this topic.